### PR TITLE
basic solution

### PR DESCRIPTION
**Pull request by IhorRome**

### CODE REVIEW

**Review by andriiromaniuk — APPROVED** [review]

File: src/main/java/mate/academy/model/User.java [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+
+@Entity
+public class User {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+    private String email;
```
Comment (MaliukDaria) [inline src/main/java/mate/academy/model/User.java:13]: lets add unique constraint to email column

File: src/main/java/mate/academy/security/AuthenticationServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+@Service
+public class AuthenticationServiceImpl implements AuthenticationService {
+    @Inject
+    private UserService userService;
+
+    @Override
+    public void register(String email, String password) throws RegistrationException {
+        Optional<User> userByEmail = userService.findByEmail(email);
+        if (userByEmail.isPresent()) {
+            throw new RegistrationException("User with this email already exists");
```
Comment (MaliukDaria) [inline src/main/java/mate/academy/security/AuthenticationServiceImpl.java:21]:
```suggestion
            throw new RegistrationException("User with this email already exists: " + email);
```

File: src/main/java/mate/academy/security/AuthenticationServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    public User login(String email, String password) throws AuthenticationException {
+        Optional<User> userFromDbOptional = userService.findByEmail(email);
+        if (userFromDbOptional.isEmpty()) {
+            throw new AuthenticationException("Wrong input email or password");
+        }
+        User user = userFromDbOptional.get();
+        if (HashUtil.hashPassword(password, user.getSalt()).equals(user.getPassword())) {
+            return user;
+        }
+        throw new AuthenticationException("Wrong input email or password");
```
Comment (MaliukDaria) [inline src/main/java/mate/academy/security/AuthenticationServiceImpl.java:36]: Can we do this only with one if statement?

File: src/main/java/mate/academy/util/HashUtil.java [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+        StringBuilder hashedPassword = new StringBuilder();
+        try {
+            MessageDigest messageDigest = MessageDigest.getInstance(CRYPTO_ALGORITHM);
+            messageDigest.update(salt);
+            byte[] digest = messageDigest.digest();
+            for (byte b : digest) {
+                hashedPassword.append(String.format("%02x", b));
+            }
+        } catch (NoSuchAlgorithmException e) {
+            throw new IllegalStateException("Can't hash password using SHA-512", e);
```
Comment (MaliukDaria) [inline src/main/java/mate/academy/util/HashUtil.java:31]:
```suggestion
            throw new IllegalStateException("Can't hash password using " + CRYPTO_ALGORITHM, e);
```

**Review by MaliukDaria — APPROVED** [review]

**Pull request closed without merging**